client/src/hooks/useDateFieldInfo.js

```diff
@@ -3,8 +3,12 @@ import { useState } from "react";
 const useDateFieldInfo = (initiallyAutocomplete = true) => {
   const [userValue, setUserValue] = useState(null);
   const [autocompleteValue, setAutocompleteValue] = useState(null);
-  const [useAutocomplete, setUseAutocompleteDefault] = useState(initiallyAutocomplete);
-  const valid = useAutocomplete ? autocompleteValue : userValue;
+  const [useAutocomplete, setUseAutocompleteDefault] = useState(
+    initiallyAutocomplete
+  );
+  const valid = useAutocomplete
+    ? Boolean(autocompleteValue)
+    : Boolean(userValue);
```

Comment:
You can probably wrap the entire ternary operator in the Boolean type conversion, e.g., `const valid = Boolean(useAutocomplete ? autocompleteValue : userValue)`